src/components/subscriptionDetail/templates/ServiceConfiguration.tsx

```diff
@@ -61,8 +61,33 @@ export const mapSplitFields = (
                     value={entry[1] !== null ? entry[1] : "null"}
                 />
             );
-        });
-    return elements.filter((entry) => entry !== null) as JSX.Element[];
+        })
+        .filter((element) => element !== null) as JSX.Element[];
+};
+
+const getVisibleInstanceFields = (
+    instance_fields: [string, any][],
+    showRelatedBlocks: boolean,
+    subBlockContainsDataForCollapsedView: boolean
+): [string, any][] => {
+    if (showRelatedBlocks || !subBlockContainsDataForCollapsedView) {
+        return [...instance_fields];
+    }
+
+    const { owner_subscription_id, subscription_instance_id, name } = instance_fields[0][1][0];
```

Comment:
I know that this is copied from the old code: maybe nice to add a bit of explanation.
So people don't need to wrap their head on `instance_fields[0][1][0]` when they read this code.

Reply:
Nicely solved